source/simulator/solver_schemes.cc

```diff
@@ -703,6 +703,180 @@ namespace aspect
+
+    // Now iterate out the nonlinearities.
+    dcr.stokes_residuals = std::pair<double,double>  (numbers::signaling_nan<double>(),
+                                                      numbers::signaling_nan<double>());
+
+    assemble_and_solve_temperature();
+    assemble_and_solve_composition();
+
+    for (nonlinear_iteration = 0; nonlinear_iteration < max_nonlinear_iterations; ++nonlinear_iteration)
+      {
+
+        assemble_and_solve_defect_correction_Stokes(dcr, true);
+
+        if (parameters.run_postprocessors_on_nonlinear_iterations)
+          postprocess ();
+
+        if (dcr.residual/dcr.initial_residual < parameters.nonlinear_tolerance)
+          break;
+      }
+
+    // Reset the linear tolerance to what it was at the beginning of the time step.
+    parameters.linear_stokes_solver_tolerance = begin_linear_tolerance;
+
+    // When we are finished iterating, we need to set the final solution to the current linearization point,
+    // because the solution vector is used in the postprocess.
+    solution = current_linearization_point;
+  }
+
+  template <int dim>
+  void Simulator<dim>::solve_iterated_advection_and_defect_correction_stokes ()
+  {
+    // Now store the linear_tolerance we started out with, because we might change
+    // it within this timestep.
+    double begin_linear_tolerance = parameters.linear_stokes_solver_tolerance;
+
+    double initial_temperature_residual = 0;
+    std::vector<double> initial_composition_residual (parameters.n_compositional_fields,0);
+
+    DefectCorrectionResiduals dcr;
+    dcr.initial_residual = 1;
+
+    dcr.velocity_residual = 0;
+    dcr.pressure_residual = 0;
+    dcr.residual = 1;
+    dcr.residual_old = 1;
+
+    dcr.switch_initial_residual = 1;
+    dcr.newton_residual_for_derivative_scaling_factor = 1;
+
+    const unsigned int max_nonlinear_iterations =
+      (pre_refinement_step < parameters.initial_adaptive_refinement)
+      ?
+      std::min(parameters.max_nonlinear_iterations,
+               parameters.max_nonlinear_iterations_in_prerefinement)
+      :
+      parameters.max_nonlinear_iterations;
+
+    // Now iterate out the nonlinearities.
+    dcr.stokes_residuals = std::pair<double,double>  (numbers::signaling_nan<double>(),
+                                                      numbers::signaling_nan<double>());
+
+
+    for (nonlinear_iteration = 0; nonlinear_iteration < max_nonlinear_iterations; ++nonlinear_iteration)
+      {
+        const double relative_temperature_residual =
+          assemble_and_solve_temperature(nonlinear_iteration == 0, &initial_temperature_residual);
+
+        const std::vector<double>  relative_composition_residual =
+          assemble_and_solve_composition(nonlinear_iteration == 0, &initial_composition_residual);
+
+        // write the residual output in the same order as the solutions
+        pcout << "      Relative nonlinear residuals (temperature, compositional fields): " << relative_temperature_residual;
```

Comment:
Although it might be possible, it would be a lot of work because currently the `assemble_and_solve_defect_correction_Stokes` is repsonible for outputing information, which includes different output for dc Picard, full Newton iterations and in the case of line search a lot of lines of output. Sot his would also change the Netwon solver output. I agree that the output should be equalized as much as possible, but I think that it is better to first get this in and discuss that in a different pull request.